src/transport/PairingSession.cpp

```diff
@@ -99,4 +100,22 @@ CHIP_ERROR PairingSession::DecodeMRPParametersIfPresent(TLV::Tag expectedTag, TL
     return tlvReader.ExitContainer(containerType);
 }
 
+void PairingSession::Clear()
+{
+    if (mSessionManager != nullptr)
+    {
+        auto handle = GetSecureSessionHandle();
+        if (handle.HasValue() && !handle.Value()->AsSecureSession()->IsActiveSession())
+        {
+            // Make sure to clean up our pending session, since we're the only
+            // ones who have access to it do do so.
+            mSessionManager->ExpirePairing(handle.Value());
+        }
+    }
+
+    mPeerSessionId.ClearValue();
+    mSecureSessionHolder.Release();
```

Comment:
This doesn't quite feel right. In general, should we not be releasing holders first _before_ we expire pairings? Otherwise `Release` is going to attempt to remove itself as a holder and access the underlying session that was just expired earlier?

Reply:
Or is it that the act of calling `ExpirePairing` will invoke the destructor for `SecureSession`, which in turn will call `NotifySessionReleased`, which in turn calls `OnSessionReleased`, resulting in the underlying session here being actually released before we even get here?

Either way, comments would be good here.

Reply:
Sigh. We honestly need more documentation elsewhere too, since these long-chain behaviors-on-destruction and the effects they have across the system are just not obvious to most layman readers, let alone someone like me who has been wading in this for some time and still take time to deduce some of these call paths.